libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/DataStore.java

```diff
@@ -172,15 +173,12 @@ public int delete(List<T> items) {
             List<Key> unprocessedItems =
                     batchWriteResult.unprocessedDeleteItemsForTable(this.table);
             if (!unprocessedItems.isEmpty()) {
-                unprocessedItems.forEach(
-                        key ->
-                                LOGGER.warn(
-                                        "Delete fail for item key: %s".formatted(key.toString())));
+                String errMessage = "Failed during batch deletion.";
+                LOGGER.error(errMessage);
+                throw new BatchDeleteException(errMessage);
             }
-            return items.size() - unprocessedItems.size();
         }
         LOGGER.info("No items to delete");
```

Comment:
This might need to be in an `else` now? Also ditto with `LogHelper`.

Reply:
done